main.js

```diff
@@ -0,0 +1,280 @@
+        string += cleanString(A[i].name, "unspecified others")
+        switch (i) {
+            case (alen - 1):
+                break;
+            case (alen - 2):
+                string += ' and ';
+                break;
+            default:
+                string += ', ';
+                break;
+        }
+    }
+
+    //console.log( string );
+
+    return string;
+}
+
+
+// Collect all countries and select relevant fields for our app
+async function getCountries() {
+    APIerror = "";
+    try {
+        const result = await axios.get('https://restcountries.eu/rest/v2/all?fields=name;capital;population;currencies;languages;flag;subregion');
+
+        for (let i in result.data) {
+            let c = {};
+            ({
+                name: c.name,
+                capital: c.capital,
+                languages: c.languages,
+                currencies: c.currencies,
+                flag: c.flag,
+                population: c.population,
+                subregion: c.subregion
+            } = result.data[i]);
+            countries.push(c);
+        }
+        //console.log( countries );
+    } catch (e) {
+        console.error(e);
+        APIerror = " Unable to collect country data. The network may be failing or the website may be down. Try again later."
+        console.warn(APIerror);
+    }
+}
+
+function changeErrorVisibility(onoff) {
+    let errorField = document.getElementById('error-field');
+
+    if (onoff) {
+        errorField.classList.remove('hidden');
+        errorField.classList.add('visible');
+    } else {
+        errorField.classList.remove('visible');
+        errorField.classList.add('hidden');
+    }
+}
+
+function showCountryError(text) {
+    let errorField = document.getElementById('error-field');
+    errorField.innerHTML = text;
+    changeErrorVisibility(true);
+    console.error(text);
+}
+
+
+// once found, show the country details as per the specification.
+// no handling of errors or omissions in the country api,
+// these issues may be transient
+
+function showCountry(countryObject) {
+    //log( "Showing " + countryObject.name );
+    document.getElementById('searchfield').value = '';
+    try {
+        delDomTree(document.getElementById('result'));
+    } catch (e) {
+        //Never mind, first time use
+    }
+    let parentdiv = document.getElementById('main');
+    let resultdiv = document.createElement('div');
+    resultdiv.id = 'result';
+
+    let flag = document.createElement('img');
+    flag.id = 'flag';
+    flag.src = countryObject.flag;
+    flag.style.height = '100px';
+
+    resultdiv.appendChild(flag);
+
+    let textdiv = document.createElement('div');
+    textdiv.id = 'description';
+
+    let nameElement = document.createElement('h2');
+    let nameElementTxt = document.createTextNode(countryObject.name);
+    nameElement.appendChild(nameElementTxt);
+    resultdiv.appendChild(nameElement);
+
+    let codestring = '';
+    codestring += countryObject.name;
```

Comment:
Deze zou je al kunnen samenvoegen: `let codestring = countyObject.name;`

Reply:
Is handiger, ik kan zo altijd += gebruiken zonder me af te vragen of dit het eerste stukje was of niet.

Reply:
Had ik in het commentaar moeten zetten.